cohortextractor/mssql_utils.py

```diff
@@ -88,3 +89,37 @@ def mssql_sqlalchemy_engine_from_url(url):
     params["drivername"] = "mssql+pyodbc"
     params["query"] = {"driver": "ODBC Driver 17 for SQL Server"}
     return sqlalchemy.create_engine(URL(**params))
+
+
+def dbapi_cursor_to_csv_file(
+    cursor, filename, batch_size=None, yield_output_rows=False
+):
+    """
+    Takes a DB-API comptabile cursor object with a result set and writes those
+    results to a CSV file
+    """
+    iterator = _dbapi_cursor_to_csv_file(cursor, filename, batch_size=batch_size)
+    # We don't want the default behaviour of this function to be to create an
+    # iterator which does nothing until you consume it (which might lead to
+    # confusion) but we do want the option of yielding the results so that
+    # callers who need to run additional checks on the output can do so
+    if yield_output_rows:
+        return iterator
+    else:
+        for line in iterator:
+            pass
```

Comment:
Given that we don't ever call `dbapi_cursor_to_csv_file` with `yield_output_rows=False`, do we need the `yield_output_rows` argument?  It's bit head-scratchy.

Reply:
Ah yeah, I just copied it back from the old `mssql_query_to_csv_file` function which used the same pattern. But you're right, there's not really any need for it.